examples/Sinusoid/emulate.jl

```diff
@@ -0,0 +1,241 @@
+plot!(inputs[1, :], inputs[2, :]; seriestype = :scatter, zcolor = outputs[2, :], label = :false)
+xlabel!("θ1")
+ylabel!("θ2")
+title!("y2")
+# Put subplots together into on figure
+p = plot(p1, p2, layout = (1, 2))
+savefig(p, joinpath(data_save_directory, "sinusoid_GP_emulator_contours.png"))
+
+
+## Random Features
+# An alternative to Gaussian process for the emulator
+# We have two input dimensions and two output dimensions.
+input_dim = 2
+output_dim = 2
+# Select number of features
+n_features = 60
+nugget = 1e-12
+kernel_structure = SeparableKernel(LowRankFactor(2, nugget), OneDimFactor())
+
+# Create random features
+optimizer_options = Dict(
+    "n_ensemble" => 20,
+    "scheduler" => EKP.DataMisfitController(on_terminate = "continue"),
+    "n_iteration" => 50,
+    "rng" => rng,
+)
+random_features = ScalarRandomFeatureInterface(
+    n_features,
+    input_dim,
+    rng = rng,
+    kernel_structure = kernel_structure,
+    optimizer_options = optimizer_options,
+)
+emulator_random_features = Emulator(random_features, input_output_pairs, normalize_inputs = true)
+optimize_hyperparameters!(emulator_random_features)
+
+# Test on new data
+y_rf, y_rf_covs = Emulators.predict(emulator_random_features, x_test_unconstrained);
+
+# Estimate root mean squared error
+rmse_rf = sqrt.(mean((y_rf - y_true) .^ 2, dims = 2))
+println("RMSE RF:", rmse_rf)     # [2.3993665360796514; 1.4247135862280327;;]
+
+# Get standard deviations from covariance matrix
+# Plots of y_true vs y_gp to check accuracy of emulator (neaten up)
+y_rf_std1 = [sqrt(y_rf_cov[1, 1]) for y_rf_cov in y_rf_covs]
+y_rf_std2 = [sqrt(y_rf_cov[2, 2]) for y_rf_cov in y_rf_covs]
+
+p1 = plot(y_true[1, :], y_gp[1, :]; yerr = y_gp_std1, seriestype = :scatter, label = "Gaussian Process")
+plot!(y_true[1, :], y_rf[1, :]; yerr = y_rf_std1, seriestype = :scatter, label = "Random Features")
+xlabel!("True y1")
+ylabel!("Predicted y1")
+Plots.abline!(1, 0; line = :dash, color = :black, label = :false)
+
+p2 = plot(y_true[2, :], y_gp[2, :]; yerror = y_gp_std2, seriestype = :scatter, label = "Gaussian Process")
+plot!(y_true[2, :], y_rf[2, :]; yerror = y_rf_std2, seriestype = :scatter, label = "Random Features")
+Plots.abline!(1, 0; line = :dash, color = :black, label = :false)
+xlabel!("True y2")
+ylabel!("Predicted y2")
+p = plot(p1, p2, layout = (1, 2))
+savefig(p, joinpath(data_save_directory, "sinusoid_GP_vs_RF_emulator.png"))
+
+# This comparison shows the GP emulator is more accurate on the test data compared to the RF emulator.
+# Both methods are more accurate around y_obs=(6.3, 7.1) with worse performance at the edges.
+# This is as expected, because our training dataset from EKI has more points focused around theta_true.
+
+# Plot outputs as contours
+# f returns emulator prediction only (no variance) given x1 and x2, for output j=1 or 2
+f(x1, x2; j = 1) =
+    Emulators.predict(emulator_random_features, Emulators.transform_constrained_to_unconstrained(prior, hcat(x1, x2)'))[1][
```

Comment:
adding the obs_noise_cov means that we now transform the data. we need to add `transform_to_real=true` (as we do with GP) in the predict methods when we calculate RMSE or plots